synapse/tools/reload.py

```diff
@@ -0,0 +1,69 @@
+import sys
+import yaml
+import asyncio
+import argparse
+
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+
+import synapse.lib.output as s_output
+import synapse.lib.urlhelp as s_urlhelp
+
+descr = '''
+List or execute reload functions on a Synapse service.
+'''
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = getArgParser()
+    opts = pars.parse_args(argv)
+
+    async with s_telepath.withTeleEnv():
+
+        async with await s_telepath.openurl(opts.svcurl) as cell:
+
+            if opts.cmd == 'list':
+                names = await cell.getReloadNames()
+                if names:
+                    outp.printf(f'Cell at {s_urlhelp.sanitizeUrl(opts.svcurl)} has the following reload functions:')
+                    for name in names:
+                        outp.printf(name)
+                else:
+                    outp.printf(f'Cell at {s_urlhelp.sanitizeUrl(opts.svcurl)} has no registered reload functions.')
+
+            if opts.cmd == 'reload':
+                outp.printf(f'Reloading cell at {s_urlhelp.sanitizeUrl(opts.svcurl)}')
+                try:
+                    ret = await cell.reloadCell(name=opts.name)
+                except Exception as e:
+                    outp.printf(f'Error reloading cell: {e}')
+                    return 1
+
+                if not ret:
+                    outp.printf('No functions executed.')
+                else:
+                    outp.printf(f'{"Name:".ljust(40)}Result:')
+                    for name, valu in ret.items():
+                        if valu:
+                            result = 'Success'
+                        else:
+                            result = 'Failed'
+                        outp.printf(f'{name.ljust(40)}{result}')
+    return 0
+
+def getArgParser():
+    pars = argparse.ArgumentParser(prog='reload', description=descr)
+    pars.add_argument('--svcurl', default='cell:///vertex/storage', help='The telepath URL of the Synapse service.')
+
+    subpars = pars.add_subparsers(required=True,
+                                  title='subcommands',
+                                  dest='cmd',)
+    pars_list = subpars.add_parser('list', help='List reload functions.')
+    reld_list = subpars.add_parser('reload', help='Reload registered functions.')
+    reld_list.add_argument('-n', '--name', type=str, help='Name of a function to reload.')
```

Comment:
"Name of a function to reload" seems like an odd way to refer to this. Perhaps "Name of a subsystem to reload" and update other public facing terms ( including the name= arg on reload() to subsystem=?